projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/KerberosAuthenticationIT.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.datajobs.it;
+
+import com.google.common.io.Files;
+import com.sun.security.auth.callback.TextCallbackHandler;
+import com.vmware.taurus.ServiceApp;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.security.kerberos.client.KerberosRestTemplate;
+import org.springframework.security.kerberos.test.KerberosSecurityTestcase;
+import org.springframework.security.kerberos.test.MiniKdc;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import java.io.File;
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+
+@TestPropertySource(properties = {
+      "datajobs.security.kerberos.enabled=true",
+      "featureflag.security.enabled=true",
+      "datajobs.security.kerberos.kerberosPrincipal=\"pa__view_foo\"",
+      "datajobs.security.kerberos.keytabFileLocation=target/foo.keytab",
+      "datajobs.security.kerberos.krb5ConfigLocation=target/krb5.conf"
+})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = ServiceApp.class)
+@ExtendWith(SpringExtension.class)
+//@AutoConfigureMockMvc
+public class KerberosAuthenticationIT {
+
+   private static KerberosSecurityTestcase kerberosSecurityTestcase;
+   private static Properties kdcConf;
+   private static File krb5;
+   private static Set<Principal> principals = new HashSet<Principal>();
+   private static File keytab;
+   private static MiniKdc kdc;
+   private static String principal = "pa__view_foo";
+
+
+//   @Autowired
+//   protected MockMvc mockMvc;
+
+   @LocalServerPort
+   int randomPort;
+
+   static {
```

Comment:
Shouldn't this be in [beforeAll](https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/BeforeAll.html)

Reply:
No since this is the only way I could think of to start the KDC before spring startup code...

Reply:
What about the Spring Extension approach that I mentioned before?

Reply:
Tried it, but I didn't find a reliable way to run the test without a static block.